### PR TITLE
Feature/add comments

### PR DESCRIPTION
**Pull request by ptrsd**

### CODE REVIEW

File: pkg/acp/client.go [hunk 1]
```diff
@@ -26,12 +29,14 @@ func NewClient(serverCertPath string, certPath string, keyPath string, cfg Confi
 	clientCertPool := x509.NewCertPool()
 	clientCertPool.AppendCertsFromPEM(clientCACert)
 
+	// It assigns a pool with certificates to the HTTP client.
```
Comment (holowinski) [inline pkg/acp/client.go:32]: Drop the `it assigns` and change to `assign a pool...` - it's gonna be more consistent with comments in other files.

File: pkg/acp/client.go [hunk 2]
```diff
@@ -26,12 +29,14 @@ func NewClient(serverCertPath string, certPath string, keyPath string, cfg Confi
...
 	if cert, err = tls.LoadX509KeyPair(certPath, keyPath); err != nil {
 		return Client{}, fmt.Errorf("could not create acp client: %w", err)
 	}
 
 	httpClient := &http.Client{
 		Transport: &http.Transport{
+			// It assigns a pool with certificates to the HTTP client.
```
Comment (holowinski) [inline pkg/acp/client.go:39]: same

File: pkg/acp/client.go [hunk 3]
```diff
@@ -10,14 +10,17 @@ import (
...
 type Client struct {
 	HttpClient *http.Client
 	config     Config
 }
 
+// NewClient creates new instance of ACP client.
 func NewClient(serverCertPath string, certPath string, keyPath string, cfg Config) (client Client, err error) {
 	var cert tls.Certificate
 
+	// It sets up the certificate HTTP client needs for TLS communication with a server.
```
Comment (holowinski) [inline pkg/acp/client.go:23]: same

File: pkg/acp/config.go [hunk 4]
```diff
@@ -7,14 +7,22 @@ import (
...
 type Config struct {
+	// RedirectURL holds information where to redirect the user after successful authentication.
 	RedirectURL string
-	ClientID    string
-	Scopes      []string
-	AuthURL     string
-	TokenURL    string
+	// ClientID is the ID of our client registered in ACP.
+	ClientID string
+	// Scopes must be at least a subset of scopes assigned to our application in ACP.
```
Comment (holowinski) [inline pkg/acp/config.go:14]: drop `at least`

File: pkg/acp/config.go [hunk 5]
```diff
@@ -7,14 +7,22 @@ import (
...
+	Scopes []string
+	// AuthURL is an URL where users can authenticate.
```
Comment (holowinski) [inline pkg/acp/config.go:16]: Hmm, that's not correct.

https://tools.ietf.org/html/rfc6749#section-3.1

File: pkg/acp/config.go [hunk 6]
```diff
@@ -7,14 +7,22 @@ import (
...
+	AuthURL string
+	// TokenURL holds information about the endpoint where we can exchange code for an access token.
+	TokenURL string
+	// PKCEEnabled is information whether PKCE is enabled or not.
```
Comment (holowinski) [inline pkg/acp/config.go:20]: `is information` is redundant

File: pkg/acp/config.go [hunk 7]
```diff
@@ -7,14 +7,22 @@ import (
...
 	PKCEEnabled bool
 }
 
+// AuthorizeURL builds the URL where the client will redirect the user after accessing /login endpoint. Challenge is a
```
Comment (holowinski) [inline pkg/acp/config.go:24]: This also seems incorrect.

**Review by holowinski — CHANGES_REQUESTED** [review]

**Review by mbilski — APPROVED** [review]

**Comment by holowinski** [comment]
Closed this one. Handled in https://github.com/cloudentity/sample-go-mtls-oauth-client/pull/7

**Pull request closed without merging**